### PR TITLE
docs: new types, conversions, builtins

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
- document new types, conversion rules, unsafe math builtins

### Cute Animal Picture

image

### CODE REVIEW

File: docs/built-in-functions.rst [hunk 1]
```diff
@@ -545,6 +545,136 @@ Math
...
 
+.. py:function:: unsafe_add(x: integer, y: integer) -> integer
+
+    Add ``x`` and ``y``, without checking for overflow. ``x`` and ``y`` must both be integers of the same type. If the result exceeds the bounds of the input type, it will be wrapped.
+
+    .. code-block:: python
+
+        @external
+        @view
+        def foo(x: uint8, y: uint8) -> uint256:
```
Comment (fubuloubu) [inline docs/built-in-functions.rst:556]: What's the difference between this and bar below?

File: docs/built-in-functions.rst [hunk 2]
```diff
@@ -545,6 +545,136 @@ Math
...
+    .. code-block:: python
+
+        >>> ExampleContract.foo(1, 1)
+        2
+
+        >>> ExampleContract.foo(255, 255)
+        254
+
+        >>> ExampleContract.bar(127, 127)
+        -2
```
Comment (fubuloubu) [inline docs/built-in-functions.rst:574]: Is this correct behavior? Shouldnt it return an unsigned integer?

File: docs/release-notes.rst [hunk 3]
```diff
@@ -3,6 +3,47 @@
 Release Notes
 #############
 
+v0.3.2
```
Comment (fubuloubu) [inline docs/release-notes.rst:6]: Usually we do this in a separate PR before release, so let's skip this for now

**Review by fubuloubu — APPROVED** [review]

File: docs/built-in-functions.rst (hunk 1 above)
Comment (charles-cooper) [reply docs/built-in-functions.rst:556]: foo accepts uint8, bar accepts int8 (signed)

File: docs/built-in-functions.rst (hunk 2 above)
Comment (charles-cooper) [reply docs/built-in-functions.rst:574]: bar takes two int8 and returns an int8

File: docs/release-notes.rst (hunk 3 above)
Comment (charles-cooper) [reply docs/release-notes.rst:6]: ok -- i'll move the commits with the release notes into a separate PR

File: docs/built-in-functions.rst (hunk 1 above)
Comment (skellet0r) [inline docs/built-in-functions.rst:556]:
```suggestion
        def foo(x: uint8, y: uint8) -> uint8:
```

File: docs/built-in-functions.rst [hunk 4]
```diff
@@ -545,6 +545,136 @@ Math
...
+    .. code-block:: python
+
+        @external
+        @view
+        def foo(x: uint8, y: uint8) -> uint256:
+            return unsafe_add(x, y)
+
+        @external
+        @view
+        def bar(x: int8, y: int8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:561]:
```suggestion
        def bar(x: int8, y: int8) -> int8:
```

File: docs/built-in-functions.rst [hunk 5]
```diff
@@ -545,6 +545,136 @@ Math
...
+
+.. py:function:: unsafe_sub(x: integer, y: integer) -> integer
+
+    Subtract ``x`` and ``y``, without checking for overflow. ``x`` and ``y`` must both be integers of the same type. If the result underflows the bounds of the input type, it will be wrapped.
+
+    .. code-block:: python
+
+        @external
+        @view
+        def foo(x: uint8, y: uint8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:587]:
```suggestion
        def foo(x: uint8, y: uint8) -> uint8:
```

File: docs/built-in-functions.rst [hunk 6]
```diff
@@ -545,6 +545,136 @@ Math
...
+            return unsafe_sub(x, y)
+
+        @external
+        @view
+        def bar(x: int8, y: int8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:592]:
```suggestion
        def bar(x: int8, y: int8) -> int8:
```

File: docs/built-in-functions.rst [hunk 7]
```diff
@@ -545,6 +545,136 @@ Math
...
+
+.. py:function:: unsafe_mul(x: integer, y: integer) -> integer
+
+    Multiply ``x`` and ``y``, without checking for overflow. ``x`` and ``y`` must both be integers of the same type. If the result exceeds the bounds of the input type, it will be wrapped.
+
+    .. code-block:: python
+
+        @external
+        @view
+        def foo(x: uint8, y: uint8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:619]:
```suggestion
        def foo(x: uint8, y: uint8) -> uint8:
```

File: docs/built-in-functions.rst [hunk 8]
```diff
@@ -545,6 +545,136 @@ Math
...
+            return unsafe_mul(x, y)
+
+        @external
+        @view
+        def bar(x: int8, y: int8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:624]:
```suggestion
        def bar(x: int8, y: int8) -> int8:
```

File: docs/built-in-functions.rst [hunk 9]
```diff
@@ -545,6 +545,136 @@ Math
...
+
+.. py:function:: unsafe_div(x: integer, y: integer) -> integer
+
+    Divide ``x`` and ``y``, without checking for division-by-zero. ``x`` and ``y`` must both be integers of the same type. If the denominator is zero, the result will (following EVM semantics) be zero.
+
+    .. code-block:: python
+
+        @external
+        @view
+        def foo(x: uint8, y: uint8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:654]:
```suggestion
        def foo(x: uint8, y: uint8) -> uint8:
```

File: docs/built-in-functions.rst [hunk 10]
```diff
@@ -545,6 +545,136 @@ Math
...
+            return unsafe_div(x, y)
+
+        @external
+        @view
+        def bar(x: int8, y: int8) -> uint256:
```
Comment (skellet0r) [inline docs/built-in-functions.rst:659]:
```suggestion
        def bar(x: int8, y: int8) -> int8:
```

**Review by skellet0r — APPROVED** [review]
:rocket:
